[Operator Versioning][Edge] Add old models and unittest

cccclai opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67731
* #67728
* #67730
* #67729
* __->__ #67726



1. Check in one model with aten:div_tensor old op with unittest in both cpp and python. The following two lines are commented out and expected to work after using upgrader.
```
_helper(mobile_module_v2, div_tensor_0_3)
_helper(current_mobile_module, torch.div)
```

2. Update the commented code accordingly.


Currently there are 6 upgraders. The following old models with operators are added to cover these 6 upgraders:
```
// Tensor x Tensor

test_versioned_div_tensor_v3


// Tensor x Scalar

test_versioned_div_scalar_float_v3
test_versioned_div_scalar_reciprocal_int_v3
test_versioned_div_scalar_inplace_float_v3

// Scalar x Scalar

test_versioned_div_scalar_scalar_v3


// Tensor x Tensor with out kwarg

test_versioned_div_tensor_out_v3


// Tensor x Tensor inplace

test_versioned_div_tensor_inplace_v3


// Tensor x Scalar inplace

test_versioned_div_scalar_inplace_int_v3

```
Note:
In this pr, per model, it includes the following test:
1. Model (with old op) load/run test will be in both cpp and python
2. Model (with old op) + upgrader test will be in python
Other tests considered adding:
1. per upgrader bytecode test
2. app level integration test

Differential Revision: [D32069653](https://our.internmc.facebook.com/intern/diff/D32069653/)

Differential Revision: [D32069653](https://our.internmc.facebook.com/intern/diff/D32069653)

Commit:
[Operator Versioning][Edge] Add old models and unittest

1. Check in one model with aten:div_tensor old op with unittest in both cpp and python. The following two lines are commented out and expected to work after using upgrader.
```
_helper(mobile_module_v2, div_tensor_0_3)
_helper(current_mobile_module, torch.div)
```

2. Update the commented code accordingly.


Note:
In this pr, per model, it includes the following test:
1. Model (with old op) load/run test will be in both cpp and python
2. Model (with old op) + upgrader test will be in python
Other tests considered adding:
1. per upgrader bytecode test
2. app level integration test

Differential Revision: [D32069653](https://our.internmc.facebook.com/intern/diff/D32069653/)

[ghstack-poisoned]

Commit:
Update on "[Operator Versioning][Edge] Add old models and unittest"


1. Check in one model with aten:div_tensor old op with unittest in both cpp and python. The following two lines are commented out and expected to work after using upgrader.
```
_helper(mobile_module_v2, div_tensor_0_3)
_helper(current_mobile_module, torch.div)
```

2. Update the commented code accordingly.


Note:
In this pr, per model, it includes the following test:
1. Model (with old op) load/run test will be in both cpp and python
2. Model (with old op) + upgrader test will be in python
Other tests considered adding:
1. per upgrader bytecode test
2. app level integration test

Differential Revision: [D32069653](https://our.internmc.facebook.com/intern/diff/D32069653/)

[ghstack-poisoned]

Commit:
Update on "[Operator Versioning][Edge] Add old models and unittest"


1. Check in one model with aten:div_tensor old op with unittest in both cpp and python. The following two lines are commented out and expected to work after using upgrader.
```
_helper(mobile_module_v2, div_tensor_0_3)
_helper(current_mobile_module, torch.div)
```

2. Update the commented code accordingly.


Note:
In this pr, per model, it includes the following test:
1. Model (with old op) load/run test will be in both cpp and python
2. Model (with old op) + upgrader test will be in python
Other tests considered adding:
1. per upgrader bytecode test
2. app level integration test

Differential Revision: [D32069653](https://our.internmc.facebook.com/intern/diff/D32069653/)

[ghstack-poisoned]

Commit:
Update on "[Operator Versioning][Edge] Add old models and unittest"




1. Check in one model with aten:div_tensor old op with unittest in both cpp and python. The following two lines are commented out and expected to work after using upgrader.
```
_helper(mobile_module_v2, div_tensor_0_3)
_helper(current_mobile_module, torch.div)
```

2. Update the commented code accordingly.


Currently there are 6 upgraders. The following old models with operators are added to cover these 6 upgraders:
```
// Tensor x Tensor

test_versioned_div_tensor_v3


// Tensor x Scalar

test_versioned_div_scalar_float_v3
test_versioned_div_scalar_reciprocal_int_v3
test_versioned_div_scalar_inplace_float_v3

// Scalar x Scalar

test_versioned_div_scalar_scalar_v3


// Tensor x Tensor with out kwarg

test_versioned_div_tensor_out_v3


// Tensor x Tensor inplace

test_versioned_div_tensor_inplace_v3


// Tensor x Scalar inplace

test_versioned_div_scalar_inplace_int_v3

```
Note:
In this pr, per model, it includes the following test:
1. Model (with old op) load/run test will be in both cpp and python
2. Model (with old op) + upgrader test will be in python
Other tests considered adding:
1. per upgrader bytecode test
2. app level integration test

Differential Revision: [D32069653](https://our.internmc.facebook.com/intern/diff/D32069653/)

Differential Revision: [D32069653](https://our.internmc.facebook.com/intern/diff/D32069653)

[ghstack-poisoned]

iseeyuan commented on test/cpp/jit/test_lite_interpreter.cpp:
```diff
@@ -1487,5 +1487,21 @@ TEST(RunTimeTest, RuntimeCall) {
 }
 
+#if !defined FB_XPLAT_BUILD
+// The following test run in fbcode only
```
Eventually it should work in xplat as well?

iseeyuan reviewed (approved):
LGTM

Commit:
Update on "[Operator Versioning][Edge] Add old models and unittest"




1. Check in one model with aten:div_tensor old op with unittest in both cpp and python. The following two lines are commented out and expected to work after using upgrader.
```
_helper(mobile_module_v2, div_tensor_0_3)
_helper(current_mobile_module, torch.div)
```

2. Update the commented code accordingly.


Currently there are 6 upgraders. The following old models with operators are added to cover these 6 upgraders:
```
// Tensor x Tensor

test_versioned_div_tensor_v3


// Tensor x Scalar

test_versioned_div_scalar_float_v3
test_versioned_div_scalar_reciprocal_int_v3
test_versioned_div_scalar_inplace_float_v3

// Scalar x Scalar

test_versioned_div_scalar_scalar_v3


// Tensor x Tensor with out kwarg

test_versioned_div_tensor_out_v3


// Tensor x Tensor inplace

test_versioned_div_tensor_inplace_v3


// Tensor x Scalar inplace

test_versioned_div_scalar_inplace_int_v3

```
Note:
In this pr, per model, it includes the following test:
1. Model (with old op) load/run test will be in both cpp and python
2. Model (with old op) + upgrader test will be in python
Other tests considered adding:
1. per upgrader bytecode test
2. app level integration test

Differential Revision: [D32069653](https://our.internmc.facebook.com/intern/diff/D32069653/)

Differential Revision: [D32069653](https://our.internmc.facebook.com/intern/diff/D32069653)

[ghstack-poisoned]

Commit:
Update on "[Operator Versioning][Edge] Add old models and unittest"




1. Check in one model with aten:div_tensor old op with unittest in both cpp and python. The following two lines are commented out and expected to work after using upgrader.
```
_helper(mobile_module_v2, div_tensor_0_3)
_helper(current_mobile_module, torch.div)
```

2. Update the commented code accordingly.


Currently there are 6 upgraders. The following old models with operators are added to cover these 6 upgraders:
```
// Tensor x Tensor

test_versioned_div_tensor_v3


// Tensor x Scalar

test_versioned_div_scalar_float_v3
test_versioned_div_scalar_reciprocal_int_v3
test_versioned_div_scalar_inplace_float_v3

// Scalar x Scalar

test_versioned_div_scalar_scalar_v3


// Tensor x Tensor with out kwarg

test_versioned_div_tensor_out_v3


// Tensor x Tensor inplace

test_versioned_div_tensor_inplace_v3


// Tensor x Scalar inplace

test_versioned_div_scalar_inplace_int_v3

```
Note:
In this pr, per model, it includes the following test:
1. Model (with old op) load/run test will be in both cpp and python
2. Model (with old op) + upgrader test will be in python
Other tests considered adding:
1. per upgrader bytecode test
2. app level integration test

Differential Revision: [D32069653](https://our.internmc.facebook.com/intern/diff/D32069653/)

Differential Revision: [D32069653](https://our.internmc.facebook.com/intern/diff/D32069653)

[ghstack-poisoned]

Commit:
Update on "[Operator Versioning][Edge] Add old models and unittest"




1. Check in one model with aten:div_tensor old op with unittest in both cpp and python. The following two lines are commented out and expected to work after using upgrader.
```
_helper(mobile_module_v2, div_tensor_0_3)
_helper(current_mobile_module, torch.div)
```

2. Update the commented code accordingly.


Currently there are 6 upgraders. The following old models with operators are added to cover these 6 upgraders:
```
// Tensor x Tensor

test_versioned_div_tensor_v3


// Tensor x Scalar

test_versioned_div_scalar_float_v3
test_versioned_div_scalar_reciprocal_int_v3
test_versioned_div_scalar_inplace_float_v3

// Scalar x Scalar

test_versioned_div_scalar_scalar_v3


// Tensor x Tensor with out kwarg

test_versioned_div_tensor_out_v3


// Tensor x Tensor inplace

test_versioned_div_tensor_inplace_v3


// Tensor x Scalar inplace

test_versioned_div_scalar_inplace_int_v3

```
Note:
In this pr, per model, it includes the following test:
1. Model (with old op) load/run test will be in both cpp and python
2. Model (with old op) + upgrader test will be in python
Other tests considered adding:
1. per upgrader bytecode test
2. app level integration test

Differential Revision: [D32069653](https://our.internmc.facebook.com/intern/diff/D32069653/)

Differential Revision: [D32069653](https://our.internmc.facebook.com/intern/diff/D32069653)

[ghstack-poisoned]

Commit:
Update on "[Operator Versioning][Edge] Add old models and unittest"




1. Check in one model with aten:div_tensor old op with unittest in both cpp and python. The following two lines are commented out and expected to work after using upgrader.
```
_helper(mobile_module_v2, div_tensor_0_3)
_helper(current_mobile_module, torch.div)
```

2. Update the commented code accordingly.


Currently there are 6 upgraders. The following old models with operators are added to cover these 6 upgraders:
```
// Tensor x Tensor

test_versioned_div_tensor_v3


// Tensor x Scalar

test_versioned_div_scalar_float_v3
test_versioned_div_scalar_reciprocal_int_v3
test_versioned_div_scalar_inplace_float_v3

// Scalar x Scalar

test_versioned_div_scalar_scalar_v3


// Tensor x Tensor with out kwarg

test_versioned_div_tensor_out_v3


// Tensor x Tensor inplace

test_versioned_div_tensor_inplace_v3


// Tensor x Scalar inplace

test_versioned_div_scalar_inplace_int_v3

```
Note:
In this pr, per model, it includes the following test:
1. Model (with old op) load/run test will be in both cpp and python
2. Model (with old op) + upgrader test will be in python
Other tests considered adding:
1. per upgrader bytecode test
2. app level integration test

Differential Revision: [D32069653](https://our.internmc.facebook.com/intern/diff/D32069653/)

Differential Revision: [D32069653](https://our.internmc.facebook.com/intern/diff/D32069653)

[ghstack-poisoned]
